vector/src/main/scala/geotrellis/vector/MultiPolygon.scala

```diff
@@ -127,10 +128,10 @@ case class MultiPolygon(jtsGeom: jts.MultiPolygon) extends MultiGeometry
   def |(ps: MultiPolygon): TwoDimensionsTwoDimensionsUnionResult =
     union(ps)
   def union(ps: MultiPolygon): TwoDimensionsTwoDimensionsUnionResult =
-    Seq(this, ps).unioned
+    (this.polygons ++ ps.polygons).toSeq.cascadeUnion
```

Comment:
Perhaps this should just be `unionGeometries` to fit with the other methods? The fact that it is a "cascading" union (not even sure what that means) might not be relevant or easy to remember to client code.